### PR TITLE
Ensure test table contains multiple active files

### PR DESCRIPTION
**Pull request by findinpath**
## Description

Ensure that the table created in the test does indeed have multiple files by partitioning the table.
Assuming that an unpartitioned table will have multiple files after being created with a CTAS statement can lead to flakiness in the tests. 

This change is related to test infrastructure and doesn't change the existing functionality of the production code.

## Related issues, pull requests, and links

Fixes https://github.com/trinodb/trino/issues/13107

### CODE REVIEW

**Comment by alexjo2144** [comment]
There's already one test with a partitioned table. Can we instead do a CTAS + INSERT to force multiple files?

File: plugin/trino-delta-lake/src/test/java/io/trino/plugin/deltalake/AbstractTestDeltaLakeCreateTableStatistics.java [hunk 1]
```diff
@@ -445,8 +445,8 @@ public void testMultiFileTable()
         try (TestTable table = new TestTable(
                 "test_partitioned_table_",
                 ImmutableList.of(),
-                ImmutableList.of(),
-                "SELECT name FROM tpch.tiny.nation UNION select name from tpch.tiny.customer",
+                ImmutableList.of("partitionkey"),
+                "SELECT 'nation' AS partitionkey, name FROM tpch.tiny.nation UNION SELECT 'customer' AS partitionkey, name FROM tpch.tiny.customer",
```
Comment (findepi) [inline plugin/trino-delta-lake/src/test/java/io/trino/plugin/deltalake/AbstractTestDeltaLakeCreateTableStatistics.java:449]: I assume UNION (should be UNION ALL, really) was used to ensure multiple data files.

should we drop the second UNION branch now?

**Comment by findinpath** [comment]
Migrated the tests to use a `CREATE TABLE` statement and `INSERT` instead of `CTAS` as before.

File: plugin/trino-delta-lake/src/test/java/io/trino/plugin/deltalake/AbstractTestDeltaLakeCreateTableStatistics.java [hunk 2]
```diff
@@ -460,38 +455,37 @@ public void testMultiFileTable()
             assertEquals(minValues.size(), minValues.stream().distinct().count());
             assertEquals(maxValues.size(), maxValues.stream().distinct().count());
         }
+        finally {
+            assertUpdate("DROP TABLE IF EXISTS " + tableName);
```
Comment (findepi) [inline plugin/trino-delta-lake/src/test/java/io/trino/plugin/deltalake/AbstractTestDeltaLakeCreateTableStatistics.java:459]: it's guaranteed to exist, so IF EXISTS is redundant

File: plugin/trino-delta-lake/src/test/java/io/trino/plugin/deltalake/AbstractTestDeltaLakeCreateTableStatistics.java [hunk 3]
```diff
@@ -437,18 +436,14 @@ public void testMultiFileTable()
...
-                .setSchema(SCHEMA)
-                .build();
-        try (TestTable table = new TestTable(
-                "test_partitioned_table_",
-                ImmutableList.of(),
-                ImmutableList.of(),
-                "SELECT name FROM tpch.tiny.nation UNION select name from tpch.tiny.customer",
-                session)) {
-            List<AddFileEntry> addFileEntries = getAddFileEntries(table.getName());
+        String tableName = "test_multifile_table_" + randomTableSuffix();
```
Comment (findepi) [inline plugin/trino-delta-lake/src/test/java/io/trino/plugin/deltalake/AbstractTestDeltaLakeCreateTableStatistics.java:439]: i must admit i don't know what testMultiFileTable() is for.
Maybe we can remove it, or add a javadoc explaining its value
cc @alexjo2144

Comment (alexjo2144) [reply plugin/trino-delta-lake/src/test/java/io/trino/plugin/deltalake/AbstractTestDeltaLakeCreateTableStatistics.java:439]: I may have known at one point, but I'm also unsure if it has any value

File: plugin/trino-delta-lake/src/test/java/io/trino/plugin/deltalake/AbstractTestDeltaLakeCreateTableStatistics.java (hunk 2 above)
Comment (findinpath) [reply plugin/trino-delta-lake/src/test/java/io/trino/plugin/deltalake/AbstractTestDeltaLakeCreateTableStatistics.java]: I switched the logic in the tests to using `TestTable` and doing an extra `INSERT` for ensuring the existence of multiple data files.

File: plugin/trino-delta-lake/src/test/java/io/trino/plugin/deltalake/AbstractTestDeltaLakeCreateTableStatistics.java (hunk 3 above)
Comment (findepi) [reply plugin/trino-delta-lake/src/test/java/io/trino/plugin/deltalake/AbstractTestDeltaLakeCreateTableStatistics.java]: if we don't know why we have a test, we don't know what's a good change and what isn't for the test prep or assertions.
let's drop it.

Comment (findinpath) [reply plugin/trino-delta-lake/src/test/java/io/trino/plugin/deltalake/AbstractTestDeltaLakeCreateTableStatistics.java]: I dropped the useless test in a separate PR.
`testMultiFileTableWithNaNValue` has been reworked to ensure that the table tested contains multiple active files.

**Review by findepi — APPROVED** [review]

**Pull request merged**